### PR TITLE
Feature/rb 142 menu de connection

### PR DESCRIPTION
**Pull request by KitetsuK**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] Norm of the code has been respected

* **In what state is your pull request?**
- [x] Ready to merge / Waiting a reviwer to see my work.
- [ ] Work In Progress (WIP) / My work is not finish but i want daily reviews. (Draft)
- [ ] CI Review / I want to see what the CI thinks of my work.
- [ ] Need feedback / Just to have feedback on what i produced. (May not be merged)

* **What kind of change does this PR introduce?** (You can choose multiple)
- [ ] Bug fix
- [x] Feature request
- [ ] New / Updated documentation
- [ ] Testing CI ( Make the pull request in draft mode)

* **What is the current behavior?** (link an issue based on the kind of change this pr introduce)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

## Summary by CodeRabbit

New Features:
- Added a new function to set the position of text in the `Raylib` namespace, enhancing user interface customization.
- Introduced a new `CREATE_LOBBY` scene and related systems, enabling users to create game lobbies.
- Added a new `sendReadyPacket` function, allowing users to signal readiness in the game.
- Introduced new functions in the `Menu` namespace for better menu handling and user interaction.

Bug Fixes:
- Removed unnecessary window closure check in the `SceneManager::run()` function, improving game loop efficiency.

Refactor:
- Expanded `SystemManagers` and `Scene` enums to accommodate new features.
- Modified `getSystemsGroups` function to return an array of size 8 instead of 6, reflecting the addition of new systems.

Style:
- Commented out code related to the `parallaxSystems` vector for cleaner code presentation.

Chores:
- Increased the maximum number of clients for the main server from 30 to 300, allowing for larger multiplayer sessions.
- Increased the maximum size allowed for lobby names and IP addresses, accommodating more diverse user inputs.

### CODE REVIEW

File: src/Client/Systems/Menus/CreateLobby/CreateLobbySystems.cpp [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        }
+
+        std::vector<std::function<void(std::size_t /*unused*/, std::size_t /*unused*/)>>
+        getCreateLobbySystems()
+        {
+            return {initCreateLobby, Systems::moveEntities};
+        }
+
+    } // namespace CreateLobby
+} // namespace Systems
```
Comment (kw4ntiq) [inline src/Client/Systems/Menus/CreateLobby/CreateLobbySystems.cpp]:
```suggestion
} // namespace Systems

```

**Review by kw4ntiq — APPROVED** [review]

File: src/Client/Systems/Graphic/GraphicSystems.cpp [hunk 2]
```diff
@@ -49,16 +49,16 @@ namespace Systems {
         std::vector<std::function<void(std::size_t, std::size_t)>> textSystems   = getTextSystems();
         std::vector<std::function<void(std::size_t, std::size_t)>> deathSystems =
             DeathSystems::getDeathSystems();
-        std::vector<std::function<void(std::size_t, std::size_t)>> parallaxSystems =
-            ParallaxSystems::getParallaxSystems();
+        //        std::vector<std::function<void(std::size_t, std::size_t)>> parallaxSystems =
+        //            ParallaxSystems::getParallaxSystems();
```
Comment (Saverio976) [inline src/Client/Systems/Graphic/GraphicSystems.cpp]: decomment

File: src/Client/Systems/Graphic/GraphicSystems.cpp [hunk 3]
```diff
@@ -49,16 +49,16 @@ namespace Systems {
...
 
 #ifndef NDEBUG
         audioSystems.insert(audioSystems.end(), debugCollisionRect);
 #endif
         audioSystems.insert(audioSystems.end(), spriteSystems.begin(), spriteSystems.end());
         audioSystems.insert(audioSystems.end(), textSystems.begin(), textSystems.end());
         audioSystems.insert(audioSystems.end(), deathSystems.begin(), deathSystems.end());
-        audioSystems.insert(audioSystems.end(), parallaxSystems.begin(), parallaxSystems.end());
+        //        audioSystems.insert(audioSystems.end(), parallaxSystems.begin(), parallaxSystems.end());
```
Comment (Saverio976) [inline src/Client/Systems/Graphic/GraphicSystems.cpp]: decomment

File: src/Client/Systems/Menus/Menu/MenuSystems.cpp [hunk 4]
```diff
@@ -93,14 +107,20 @@ namespace Systems {
...
             Registry::components<Types::InputBox> arrInputBox =
                 Registry::getInstance().getComponents<Types::InputBox>();
             std::vector<std::size_t> ids = Registry::getInstance().getEntitiesByComponents(
                 {typeid(Types::InputBox), typeid(Raylib::Text)});
+            static auto clockId = Registry::getInstance().getClock().create();
 
             for (auto id : ids) {
-                if (arrInputBox[id].selected && Raylib::isKeyPressed(Raylib::KeyboardKey::KB_BACKSPACE)) {
-                    deleteInputBoxChar(id, arrInputBox);
+                if (arrInputBox[id].selected && Raylib::isKeyDown(Raylib::KeyboardKey::KB_BACKSPACE)) {
```
Comment (Saverio976) [inline src/Client/Systems/Menus/Menu/MenuSystems.cpp]: if before check

File: assets/Json/menu.json [hunk 5]
```diff
@@ -26,39 +27,56 @@
...
             ]
         },
-        "ip": {
+        {
             "name": "ip",
             "type": "input",
             "font": "assets/Fonts/soliden/SolidenTrial-Black.otf",
-            "spritePath": "assets/R-TypeSheet/r-typesheet1.gif",
+            "spritePath": "assets/Menu/input-box.png",
             "text": "localhost",
```
Comment (Saverio976) [inline assets/Json/menu.json]:
```suggestion
            "text": "",
```

**Review by Saverio976 — CHANGES_REQUESTED** [review]

**Comment by Saverio976** [comment]
Déjà sur https://github.com/X-R-G-B/R-Bus/pull/143

**Pull request closed without merging**